rando_modules/modify_itempool.py

```diff
@@ -1,318 +1,123 @@
+        if pool_score < target_score:
+            # Upgrade an item
+            item_weights = [highest_score - item["score"] for item in new_items]
+            i, = random.choices([i for i in range(len(new_items))], item_weights)
+            old_item = new_items[i]
+            legal_items = [item for item in item_scores if item["score"] > old_item["score"] and item["type"] == old_item["type"]]
         else:
-            # Determine item type to pick
-            rnd_value = random.random() * 100
-            probability_count = 0
-            chosen_type = TYPE_BATTLEITEM # default
-            for item_type, chance in itemtype_chances.items():
-                probability_count += chance
-                if rnd_value <= probability_count:
-                    chosen_type = item_type
-                    break
+            # Downgrade an item
+            item_weights = [item["score"] - lowest_score for item in new_items]
+            i, = random.choices([i for i in range(len(new_items))], item_weights)
+            old_item = new_items[i]
+            legal_items = [item for item in item_scores if item["score"] < old_item["score"] and item["type"] == old_item["type"]]
+
+        # If there's no legal items to upgrade/downgrade to, try again
+        if len(legal_items) == 0:
+            continue
 
-            # Search for lower tier item of the chosen item type
-            item_found = False
-            new_item_id = 0
-            while not item_found:
-                if (new_scarcity_tier >= 1
-                and scarcity_tiers.get(new_scarcity_tier).get(chosen_type)
-                ):
-                    new_item_id = random.choice(scarcity_tiers.get(new_scarcity_tier).get(chosen_type))
-                    item_found = True
-                elif new_scarcity_tier >= 1:
-                    new_scarcity_tier = new_scarcity_tier - 1
-                else:
-                    # Reached tier 0: Coin only
-                    new_item_id = scarcity_tiers.get(new_scarcity_tier)
-                    item_found = True
+        new_item = random.choice(legal_items)
+        pool_score += new_item["score"] - old_item["score"]
+        new_items[i] = new_item
 
-            # Add item
-            new_item = Item.get(Item.value == new_item_id)
-            new_itempool.append(new_item)
+    return new_items
 
-            #print(f"Changed {item_obj} to {new_item}")
+def get_randomized_itempool(itempool:list, consumable_mode:int, scarcity:int) -> list:
+    """
+    Returns a randomized general item pool according to consumable mode
+    Balanced random mode creates an item pool that has a value equal
+    to the input pool's value, multiplied by the scarcity percentage
+    """
+    # Consumable mode:
+    # 0: vanilla (no scarcity)
+    # 1: full random (no scarcity)
+    # 2: balanced random (scarcity applies)
+    # 3: mystery only (no scarcity)
+
+    # vanilla
+    if consumable_mode == 0:
```

Comment:
Ice recently merged a refactor that uses enums accross the board, so it would be better if you can merge the latest dev in your branch and add an enum for these values in enum_options.py